### PR TITLE
Add scheduled testgrid jobs feature

### PR DESCRIPTION
**Pull request by KosalaHerath**
**Purpose**

Increase efficiency of Test Grid and reduce the cost for test running process.

**Goals**

- Reduce cost for test running process.
- Add more flexibility to test scheduling.
- Introduce new test scheduling method.
- Automate test running process.

**Approach**

Introduce new scheduling mechanism with manual, daily, weekly and monthly scheduling options. For each scheduling option new combination generate algorithm has been introduced. After reading the configurations from jobconfig file, the needed schedule is decided and according to that test plans are generated. 

**User stories**

N/A

**Release note**

N/A

**Documentation**

N/A

**Training**

N/A

**Certification**

N/A

**Marketing**

N/A

**Automation tests**
 - Unit tests 
   N/A
 - Integration tests
   N/A

**Security checks**
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

**Samples**

N/A

**Related PRs**

N/A

**Migrations (if applicable)**

N/A

**Test environment**

N/A
 
**Learning**

N/A

### CODE REVIEW

File: common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kasunbg) [inline common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java]: license year is 2019

File: common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    /**
+     * Describes a given build within a job configuration.
+     * A build config should describe its schedule,
+     * and what combinations are there that can be tested.
+     */
+    public static class Build implements Serializable {
+
+        private static final long serialVersionUID = 8474784500779659241L;
+
+        private String schedule;
```
Comment (kasunbg) [inline common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java]: What's the default schedle if the user does not specify one?

Comment (KosalaHerath) [reply common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java]: Default schedule is parameterized in TestGridConstants.java file and it is denied as "manual".

File: common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java (hunk 1 above)
Comment (KosalaHerath) [reply common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java]: Changed as 2019.

File: web/src/main/react-dashboard/package-lock.json [hunk 3]
```diff
@@ -4,36 +4,23 @@
   "lockfileVersion": 1,
```
Comment (kasunbg) [inline web/src/main/react-dashboard/package-lock.json]: Shall we remove these changes?

Comment (KosalaHerath) [reply web/src/main/react-dashboard/package-lock.json]: Removed.

**Pull request merged**